### ISSUE TITLE
Changes to y value of sprite component of NPCs not respected

### ISSUE THREAD

**Issue by Feenick** [ohrrpgce/ohrrpgce#1203]
`put slice(get npc sprite (npcRef),varI*4,varI*4)`, for example, only seems to move the sprite component of the npc by its x value, with its y staying at 0.

Strangely enough, it seems to be happening at the start of a tick; show string() with a string looking at that value seems to show the expected y value, and moving around the sprite component in the slice editor has it display as normal, only for it to snap back the moment I exit out.

[Example file [here](https://www.dropbox.com/s/5ai4tr14ztzswsx/npc%20sprite%20move%20bug.zip?dl=0); the script on the NPC is meant to move it by the same value in both the x and y directions]

**Comment by rversteegen** [comment]
Yes, that's because positions of certain slices are updated (set) immediately before the screen is redrawn. The sprite Y position is set based on the NPC/hero Z and foot offset. Walkabout slice positions are updated to NPC/hero positions at the same time.

This does need improving though, it's a big nuisance. Changing the position of the walkabout contain slice or sprite component ought to immediately modify the hero/NPC X/Y/Z to match.

**Comment by rversteegen** [comment]
So for now the workaround is
```
put slice(get npc sprite (npcRef),varI*4,varI*4)
set npc Z(npcRef, varI * -4)
```

**Comment by Feenick** [comment]
I did eventually figure out that workaround, but thanks anyway.

The thing is, I can imagine some situations where you'd want to move around the sprite component of an NPC while also keeping their x/y position intact. Maybe it should be stricter still [not allowing the x component of the sprite to be offset either], since only having the x offset being changeable via put slice() is a bit wonky.
